### ISSUE TITLE
Office365, Github and execd Integration tests fails to provision Windows agent

### ISSUE THREAD

**Issue by Rebits** [wazuh/wazuh#22503]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0-beta4 | Office365/Github/Execd | Agent | Packages | - |

### Description

It has been detected in [Release 4.8.0 - Beta 4 - Integration tests](https://github.com/wazuh/wazuh/issues/22479) that Office365, Github, and execd integration GA tests fail in the installation of the Windows agent step.

```
.\wazuh-agent--.msi: D:\a\_temp\0d0710ad-b0dd-481a-841e-ad42dd1bdd6b.ps1:3
Line |
   3 |  .\wazuh-agent--.msi /q WAZUH_MANAGER="127.0.0.1"
     |  ~~~~~~~~~~~~~~~~~~~
     | The term '.\wazuh-agent--.msi' is not recognized as a name of a cmdlet, function, script file, or executable
     | program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
     | again.
```

This is probably related to a bad handling of the agent version in case of using a tag instead of branch

### Evidences
- **Job**: https://github.com/wazuh/wazuh/actions/runs/8264472232/job/22608313463

**Comment by Selutario** [comment]
I'm moving this issue to the core project since those integrations are written in C.

**Comment by nmkoremblum** [comment]
Hi, were these tests ever run before?

Additionally, from what I can see, the tests are not written in C but in YAML:

https://github.com/wazuh/wazuh/blob/6f672d01053d274eab8f6d5622e995f6a588f6f8/.github/workflows/integration-tests-office365-tier-0-1-win.yml#L77-L81

**Comment by nmkoremblum** [comment]
Aren't the `agentd`, `sca` and `syscollector` tests failing?

**Comment by nmkoremblum** [comment]
The source of the problem has been reported here: https://github.com/wazuh/wazuh/issues/22424

**Comment by cborla** [comment]
This issue is repeated from:
- https://github.com/wazuh/wazuh/issues/22424

In which it was decided to implement the fix for 4.9.0. due to which this issue is closed.
